### ISSUE TITLE
np.where not documented to broadcast its inputs

### ISSUE THREAD

**Issue by anntzer** [numpy/numpy#7914]
The docs for `where` state

```
    condition : array_like, bool
        When True, yield `x`, otherwise yield `y`.
    x, y : array_like, optional
        Values from which to choose. `x` and `y` need to have the same
        shape as `condition`.
```

Actually a quick test shows that at least 0-d `x`/`y` inputs are broadcasted accordingly.

**Comment by eric-wieser** [comment]
Related: #8599

**Comment by eric-wieser** [comment]
Fixed in aa746c68d092022df0952e43b29ca4abe092d03b
